video/out/wayland_common.c

```diff
@@ -918,10 +919,24 @@ static void handle_toplevel_config(void *data, struct xdg_toplevel *toplevel,
 
     if (!is_fullscreen && !is_maximized) {
         if (vo_opts->keepaspect) {
-            double scale_factor = (double)width / wl->reduced_width;
-            width = ceil(wl->reduced_width * scale_factor);
+            double scale_factor_w = (double)width / wl->reduced_width;
+            double scale_factor_h = (double)height / wl->reduced_height;
```

Comment:
Instead of calculating both of these, how about checking horizontal earlier with something like this?
```
bool horizontal = abs(wl->toplevel_width - old_toplevel_height) >= abs(wl->toplevel_height - old_toplevel_height)
```

Then you can just do the swap afterwards if needed and only need one scale_factor.

Reply:
That's what I tried first, but then it switches between horizontal and vertical resizing because the cursor moved more in one direction for one update, and then switches to the other one the next update, resulting in the window jumping around a lot.

Reply:
Ah yeah good point.